models/optional_modules/core_web_vitals/snowflake/snowplow_web_vital_measurements.sql

```diff
@@ -0,0 +1,74 @@
+{{
+  config(
+    materialized='table',
+    enabled=var("snowplow__enable_cwv", false) and target.type == 'snowflake' | as_bool(),
+    sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt'))
+  )
+}}
+
+with measurements as (
+
+  select
+    page_url,
+    device_class,
+    geo_country,
+    cast( {{ dbt.date_trunc('day', 'derived_tstamp') }} as {{ dbt.type_string() }}) as time_period,
+    count(*) as page_view_count,
+    grouping_id(page_url, device_class) as id_url_and_device,
+    grouping_id(device_class) as id_device,
+    grouping_id(cast( {{ dbt.date_trunc('day', 'derived_tstamp') }} as {{ dbt.type_string() }})) as id_period,
+    grouping_id(cast( {{ dbt.date_trunc('day', 'derived_tstamp') }} as {{ dbt.type_string() }}), device_class) as id_period_and_device,
+    grouping_id(geo_country) as id_country,
+    grouping_id(geo_country, device_class) as id_country_and_device,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by lcp) as lcp_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by fid) as fid_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by cls) as cls_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by ttfb) as ttfb_{{ var('snowplow__cwv_percentile') }}p
+  from {{ ref('snowplow_web_vitals') }}
+
+  where cast(derived_tstamp as date) >= {{ dateadd('day', '-'+var('snowplow__cwv_days_to_measure')|string, date_trunc('day', snowplow_utils.current_timestamp_in_utc())) }}
+
+  group by grouping sets ((), (page_url, device_class), (device_class), (cast( {{ dbt.date_trunc('day', 'derived_tstamp') }} as {{ dbt.type_string() }})), (cast( {{ dbt.date_trunc('day', 'derived_tstamp') }} as {{ dbt.type_string() }}), device_class), (geo_country), (geo_country, device_class))
+
+)
+
+, measurement_type as (
+
+  select
+    *,
+    case when id_url_and_device <> 0 and id_device <> 0 and id_period <> 0 and id_period_and_device <> 0 and id_country <> 0 and id_country_and_device <> 0 then 'overall'
+       when id_url_and_device = 0 then 'by_url_and_device'
+       when id_period_and_device = 0 then 'by_day_and_device'
+       when id_country_and_device = 0 then 'by_country_and_device'
+       when id_country = 0 then 'by_country'
+       when id_device = 0 then 'by_device'
+       when id_period = 0 then 'by_day'
+       end as measurement_type,
+   {{ snowplow_web.core_web_vital_results('_' + var('snowplow__cwv_percentile') | string + 'p') }}
+
```

Comment:
I think the ~ in dbt is meant to do the safe string casting for you, but this is fine.